guidance/models/_azure_openai.py

```diff
@@ -43,7 +49,21 @@ def __init__(
         compute_log_probs=False,
         **kwargs,
     ):
-        """Build a new AzureOpenAI model object that represents a model in a given state."""
+        """Build a new AzureOpenAI model object that represents a model in a given state.
+
+        Parameters
+        ----------
+        model : str
+            The name of the OpenAI model to use (e.g. gpt-3.5-turbo).
```

Comment:
Are we assuming users will use the openAI naming scheme when supplying this info, even if their deployment is named something different?

Reply:
They have to. This is supposed to be the name in the drop down when they deploy the model:
![image](https://github.com/guidance-ai/guidance/assets/28632930/f2776ee6-52ad-428f-a439-7b4a7a8be3f4)